pkg/daemon/discover/discover.go

```diff
@@ -76,6 +77,10 @@ func Run(context *clusterd.Context, probeInterval time.Duration) error {
 		select {
 		case <-sigc:
 			logger.Infof("shutdown signal received, exiting...")
+			err := releaseLVMDevices(context)
```

Comment:
The change here is freeing up the LVs when the discover pod is shutting down. Is this for a scenario when the node is going down? In this case, won't there be a race condition with OSD pods on the same node that are shutting down? If the OSD pods shut down after the discover pod, the LVs wouldn't be freed. 

Let's clarify what scenarios we are trying to solve here. I was thinking of two core scenarios:
1. When a cluster is uninstalled, ensure that the LVs are all freed so the PVs can be deleted 
1. When an OSD pod goes down (or the whole node), free up the LVs so the PV can be moved to another node (if portable=true)

For scenario #1, ideally we could solve this with the [shutdown in daemon.go](https://github.com/rook/rook/pull/3755/files#diff-d1ca60bbccae22948f1847a6b9747cd5L83) that would clean up after the OSD process exits. Something else to try is with a sidecar that would watch for the osd process to exit, but i'm not sure that will help us any more during the shutdown sequence than what you already tried. An alternative cleanup could be for the operator to start a cleanup job on nodes with OSDs to free the  LVs when the cluster is removed.